pkg/storage/stores/local/downloads.go

```diff
@@ -22,6 +22,9 @@ func (s *Shipper) checkStorageForUpdates(ctx context.Context, period string, fc
 		return
 	}
 
+	fc.RLock()
+	defer fc.RUnlock()
+
 	// listing tables from store
 	var objects []chunk.StorageObject
 	objects, _, err = s.storageClient.List(ctx, period+"/")
```

Comment:
This is for single binary mode only to avoid downloading files which we already have. I have removed it since we are anyways going to cleanup files from active index directory(where ingesters write the files) to avoid keeping them forever.